File: Week6_Strings/palindromic-substrings.java
class Solution {
    public int expandFromCtr(String s, int l, int r) {
        int count = 0;
        while (l >= 0 && r < s.length()) {
            if (s.charAt(l) == s.charAt(r)) {
                count++; l--; r++;
            }
            else break;
        }   
        return count;
    }
    
    public int countSubstrings(String s) {
        int totalCount = 0;
        for (int i = 0; i < s.length(); i++) {
            totalCount += expandFromCtr(s, i, i);
            totalCount += expandFromCtr(s, i, i + 1);
        }
        return totalCount;
    }
}

File: Week6_Strings/palindromic-substrings.py
class Solution:
    def expandFromCtr(self, s, l, r):
        count = 0
        while l >= 0 and r < len(s):
            if s[l] == s[r]:
                count += 1
                l -= 1
                r += 1
            else:
                break
        return count
                
    
    def countSubstrings(self, s: str) -> int:
        totalCount = 0
        for i in range(len(s)):
            totalCount += self.expandFromCtr(s, i, i)
            totalCount += self.expandFromCtr(s, i, i + 1)
        return totalCount